File: Chris/2022-06-17-ContainerWithMostWater.cs
public class Solution {
    public int MaxArea(int[] height) {
        
        int r = height.Length - 1;
        int l = 0;
        int maxArea = 0;
        
        
        while(r > l){
            maxArea = Math.Max(maxArea, Math.Min(height[l],height[r])*(r-l));
            if(height[r] > height[l]){
                l++;
            } else {
                r--;
            }
        }
        
        return maxArea;
        
        
    }
}

File: Chloe Gwon/2022-06-17-Container-With-Most-Water.js
var maxArea = function(height) {
    //O(n)
    
    var result = 0;
    var r = height.length - 1;
    var l = 0;
    
    while (l < r){
        var area = (r-l) * Math.min(height[l], height[r]);
        if (height[l] < height[r]){
            l++;
        } else {
            r--;
        }
        result = Math.max(area, result);
    }
    
    return result;
};
